sqlalchemy_risingwave/base.py

```diff
@@ -93,25 +75,37 @@ def get_columns(self, conn, table_name, schema=None, **kw):
             name, type_str = row[:2]
             # When there are type parameters, attach them to the
             # returned type object.
-            m = re.match(r"^(\w+(?: \w+)*)(?:\(([0-9, ]*)\))?$", type_str)
-            if m is None:
-                warn("Could not parse type name '%s'" % type_str)
-                typ = sqltypes.NULLTYPE
-            else:
-                type_name, type_args = m.groups()
+            m = re.match(r"^struct<([a-z ]+)>$", type_str)
+            if m:
+                sub_type = m.group(1)
                 try:
-                    type_class = _type_map[type_name.lower()]
+                    type_class = _type_map[sub_type]
+                    warn("Struct is not supported")
+                    type_class = sqltypes.NULLTYPE
                 except KeyError:
-                    warn(f"Did not recognize type '{type_name}' of column '{name}'")
+                    warn(f"Did not recognize type '{sub_type}' of column '{name}'")
                     type_class = sqltypes.NULLTYPE
-                if type_args:
-                    typ = type_class(*[int(s.strip()) for s in type_args.split(",")])
+            else:
+                m = re.match(r"^([a-z ]+)((\[\])*)$", type_str)
+                if m:
+                    groups = m.groups()
+                    type_name = groups[0]
+                    try:
+                        type_class = _type_map[type_name]
+                        if len(groups) > 1:
+                            array_suffixs = re.findall(r"\[\]", groups[1])
+                            for i in range(0, len(array_suffixs)):
```

Comment:
Write some comments for how the type str looks like? And IIRC seems like Array only support one type.

Reply:
RisingWave supports multi-dimensional array so I write this. We can also refuse multi-dimensional array.